### DISCUSSION TITLE
Can't Interact With Calendar

### DISCUSSION THREAD

**Discussion by myalexbennett** [nhn/tui.calendar#1333] [Q&A] [answered]
## Version
2.1.3

## Test Environment
Chrome, Windows 10, JavaScript Version

## Current Behavior
The Edit, Move, Create, and Delete functions don't work.  Whenever I try to modify the calendar using the default popups or drag and drop it never saves to the calendar.

These feature don't seem to be working on the example month view either.

```js
        const Calendar = tui.Calendar;

        const calendar = new Calendar('#calendar', {
            defaultView: 'month',
        });

        calendar.setOptions({
            useFormPopup: true,
            useDetailPopup: true,
            useCreationPopup: true,
            isReadOnly: false,
        });

        calendar.setCalendars([
            {
                id: 'cal1',
                name: 'Genoa',
                backgroundColor: '#03bd9e',
            },
            {
                id: 'cal2',
                name: 'Turin',
                backgroundColor: '#00a9ff',
            },
            {
                id: 'cal3',
                name: 'Jay',
                backgroundColor: '#FFA500',
            },
        ]);
```

## Expected Behavior
Do the popups work by default or do I have to do something to enable them?

**Comment by adhrinae** [comment, marked as answer]
You need to register event handlers when you create, edit, and delete events.

Like this example below:

https://github.com/nhn/tui.calendar/blob/main/apps/calendar/examples/scripts/app.js#L192-L214

Check out [the documentation](https://github.com/nhn/tui.calendar/blob/main/docs/en/apis/calendar.md#instance-events) for further information.

**Reply by triumphhub** [reply]
Thanks for the awesome software. I too seem to have the same issue. I've spent several hours reading your links and the documentation but still cannot get calendar events to accept any modification. I'm using the "tui.calendar-main\apps\calendar\examples\15-template-popup.html" to test and although I can delete events, the move or edit events do not save to the calendar. I'm showing no console errors.
Any help would be appreciated.  I am on Windows 10 and using the most up-to-date Chrome browser.

**Reply by adhrinae** [reply]
That example doesn't include any modification for events. It's only showing how you can apply templates to popups.

Have you tried to register `beforeCreateEvent`, `beforeUpdateEvent`, and `beforeDeleteEvent` to your calendar instance like the link above?

**Reply by triumphhub** [reply]
Thank you for clarifying.
